src/Facility.Core/Http/ServiceHttpHandler.cs

```diff
@@ -180,14 +215,156 @@ protected ServiceHttpHandler(ServiceHttpHandlerSettings? settings)
+		if (eventResult.IsFailure)
+		{
+			var error = eventResult.Error!;
+			context.Result = ServiceResult.Failure(error);
+			httpResponse = await CreateHttpResponseForErrorAsync(error, httpRequest).ConfigureAwait(false);
+		}
+		else
+		{
+			if (mapping.ResponseMappings.Count != 1)
+				throw new InvalidOperationException($"Expected exactly one response mapping for {typeof(TResponse).Name}.");
+
+			var responseMapping = mapping.ResponseMappings[0];
+			httpResponse = new HttpResponseMessage(responseMapping.StatusCode)
+			{
+				Content = new EventStreamHttpContent<TResponse>(eventResult.Value, responseMapping, m_contentSerializer, m_skipResponseValidation),
+				RequestMessage = httpRequest,
+			};
 		}
 
-		httpResponse.RequestMessage = httpRequest;
 		await AdaptTask(ResponseReadyAsync(httpResponse, cancellationToken)).ConfigureAwait(true);
 
 		return httpResponse;
 	}
 
+	private sealed class EventStreamHttpContent<TResponse> : HttpContent
+		where TResponse : ServiceDto, new()
+	{
+		public EventStreamHttpContent(IAsyncEnumerable<ServiceResult<TResponse>> enumerable, HttpResponseMapping<TResponse> responseMapping, HttpContentSerializer contentSerializer, bool skipResponseValidation)
+		{
+			m_enumerable = enumerable;
+			m_responseMapping = responseMapping;
+			m_contentSerializer = contentSerializer;
+			m_skipResponseValidation = skipResponseValidation;
+
+			Headers.ContentType = new MediaTypeHeaderValue("text/event-stream");
+		}
+
+		protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context) =>
+			DoSerializeToStreamAsync(stream, CancellationToken.None);
+
+#if NET6_0_OR_GREATER
+		protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context, CancellationToken cancellationToken) =>
+			DoSerializeToStreamAsync(stream, cancellationToken);
+#endif
+
+		private async Task DoSerializeToStreamAsync(Stream stream, CancellationToken cancellationToken)
+		{
+			await foreach (var result in m_enumerable.WithCancellation(cancellationToken).ConfigureAwait(false))
+			{
+				object dto;
+				var isError = false;
+				var shouldStop = false;
+
+				if (result.IsSuccess)
+				{
+					var response = result.Value;
+					if (!m_skipResponseValidation && !response.Validate(out var responseErrorMessage))
+					{
+						dto = ServiceErrors.CreateInvalidResponse(responseErrorMessage);
+						isError = true;
+						shouldStop = true;
+					}
+					else
+					{
+						dto = m_responseMapping.GetResponseBody(response)!;
+					}
+				}
+				else
+				{
+					dto = result.Error!;
+					isError = true;
+				}
+
+				if (isError)
+				{
+#if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
+					await stream.WriteAsync(s_errorEventLine, cancellationToken).ConfigureAwait(false);
+#else
+					await stream.WriteAsync(s_errorEventLine.ToArray(), 0, s_errorEventLine.Length, cancellationToken).ConfigureAwait(false);
+#endif
```

Comment:
Defining an extension method (for `WriteAsync` on old frameworks) that takes `Memory<T>` and uses `MemoryMarshal.TryGetArray` (or `ToArray`, but I assume the former should work) can avoid needing to sprinkle all these `#if` through the rest of the code.

Reply:
Example: https://github.com/mysql-net/MySqlConnector/blob/e2a0f5884714ee588a48440872450fdedb133b42/src/MySqlConnector/Utilities/Utility.cs#L487-L511